src/lib/support/tests/TestPool.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2016-2017 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Unit tests for the Chip Pool API.
+ *
+ */
+
+#include "TestSupport.h"
+
+#include <set>
+
+#include <support/Pool.h>
+#include <support/TestUtils.h>
+
+#include <nlunit-test.h>
+
+namespace chip {
+
+template <class T, size_t N>
+template <typename F>
+void BitMapObjectPool<T, N>::ForEachActiveObject(F f)
+{
+    for (size_t word = 0; word * kBitChunkSize < N; ++word)
+    {
+        auto & usage = mUsage[word];
+        auto value   = usage.load();
```

Comment:
There is no use to synchronizing here.

load(std::memory_order_relaxed)

Reply:
Done